test/systemtest/utils.py

```diff
@@ -367,6 +367,7 @@ def is_binary_string(bytes):
 
 def load_sw_over_spi(tmp_path,
                      spiflash_path,
+                     spiflash_bin_arg,
```

Comment:
This function is now modified to work with both spiflash tool variants by assuming some uniformity in the command line arguments between the two tools, and passing others in through spiflash_bin_arg. I find that a bit too fragile.

Let's either have two functions load_sw_over_spi_with_xxx(), or add a argument that explicitly chooses one or the other tool variant.

(Or make the command line of both tools really identical to make them drop-in replacements.)

Reply:
Yes, I agree. This isn't ideal. I've now changed the implementation to generate the required argument based on `spiflash_path`. This way, the calling function just specifies the loader tool and the binary (+ optional arguments). The rest is handled by `load_sw_over_spi()`. Having another argument for the tool selection could lead to even more confusion.